File: app/src/main/java/com/stedi/randomimagegenerator/app/presenter/interfaces/SimpleIntegerParamsPresenter.java
package com.stedi.randomimagegenerator.app.presenter.interfaces;

import com.stedi.randomimagegenerator.app.presenter.interfaces.core.Presenter;
import com.stedi.randomimagegenerator.app.presenter.interfaces.core.UI;

public interface SimpleIntegerParamsPresenter extends Presenter<SimpleIntegerParamsPresenter.UIImpl> {
    void getValues();

    boolean canBeRandom();

    boolean setRandomValue();

    boolean setValue(int value);

    interface UIImpl extends UI {
        void showRandomValue();

        void showValue(int value);

        void showErrorIncorrectValue();
    }
}

File: app/src/main/java/com/stedi/randomimagegenerator/app/presenter/interfaces/SimpleIntegerParamsPresenter.kt
package com.stedi.randomimagegenerator.app.presenter.interfaces

import com.stedi.randomimagegenerator.app.presenter.interfaces.core.Presenter
import com.stedi.randomimagegenerator.app.presenter.interfaces.core.UI

interface SimpleIntegerParamsPresenter : Presenter<SimpleIntegerParamsPresenter.UIImpl> {
    fun getValues()

    fun canBeRandom(): Boolean

    fun setRandomValue()

    fun setValue(value: Int): Boolean

    interface UIImpl : UI {
        fun showRandomValue()

        fun showValue(value: Int)

        fun showErrorIncorrectValue()
    }
}